stretch_core/stretch_core/joint_trajectory_server.py

```diff
@@ -245,7 +260,15 @@ def execute_trajectory(self, goal_handle):
             self.node.get_logger().info(
                 f'New follow_joint_trajectory goal with {n_points} points, {n_joints} joints over {dt} seconds.')
 
-            # TODO: Add the waypoints
+            for index, name in enumerate(goal.trajectory.joint_names):
+                t_comp = self.trajectory_components[name]
+
+                # Set Initial waypoint
+                goal.trajectory.points[0].positions[index] = t_comp.get_position()
+                if index < len(goal.trajectory.points[0].velocities):
```

Comment:
Why are we checking this statement for velocity but not position?

Reply:
The check on line 235 ensures that the position is defined for each joint name.

Reply:
Thanks!